### PR TITLE
Implement abstract part of model

### PR DESCRIPTION
**Pull request by vismate**
I implemented the abstract building blocks of the model. @AMIOL7  and @bgdxqi,  please review and let me know what you think!

### CODE REVIEW

File: app/src/main/java/Model/Graph.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * Represents a composition of {@link Model.Node Nodes}. Notice that Graph
+ * itself inherits from {@link Model.Node Node}, therefore subgraphs (nested
+ * graphs) are permitted.
+ *
+ * @author Máté Visnyár
+ */
+public class Graph extends Node {
+
+    private ArrayList<Node> nodes;
+    private boolean isCollapsable;
```
Comment (AMIOL7) [inline app/src/main/java/Model/Graph.java]: Maybe it would be better to make this field final? I do not see any possibility when it can change

**Comment by AMIOL7** [comment]
@vismate I have left a comment to one of the files

**Comment by vismate** [comment]
> @vismate I have left a comment to one of the files

@AMIOL7 
I can't seem to find your comment. I have 'show comments' turned on. Are you sure you actually sent it?

**Comment by AMIOL7** [comment]
I marked it in the file. Maybe it is called "change"? Anyway, it was about making the isCollapsible field of graph final @vismate

**Comment by vismate** [comment]
> I marked it in the file. Maybe it is called "change"? Anyway, it was about making the isCollapsible field of graph final @vismate

Still can't see it... nevermind, we'll figure this out. I also left in an unused import, so I'll fix these ASAP.

**Comment by vismate** [comment]
Fixed problems mentioned above in [latest commit](https://github.com/AMIOL7/ELME/commit/7eebad4c82a27a98ee2a7f89fb0c5f4933e73f23).
@AMIOL7 and @bgdxqi please review the changes, so we can move forward merging the branch!

**Review by AMIOL7 — APPROVED** [review]
I see no further things that need correcting

**Review by bgdxqi — APPROVED** [review]
Looks good to me!

**Pull request merged**